pkg/client/fake/client.go

```diff
@@ -1108,12 +1108,30 @@ func (sw *fakeSubResourceClient) Create(ctx context.Context, obj client.Object,
 func (sw *fakeSubResourceClient) Update(ctx context.Context, obj client.Object, opts ...client.SubResourceUpdateOption) error {
 	updateOptions := client.SubResourceUpdateOptions{}
 	updateOptions.ApplyOptions(opts)
-
-	body := obj
-	if updateOptions.SubResourceBody != nil {
-		body = updateOptions.SubResourceBody
+	gvr, err := getGVRFromObject(obj, sw.client.scheme)
```

Comment:
Hey @troy0820 , thanks so much for jumping into this and providing a fix this quickly!

The root of the problem seems to be that `copyNonStatusFrom` is basically broken, because it is not going to clear fields that exist on the status object but not on the main object. Rather than working around that like you suggested, I'd prefer it if we kept the status handling in one place and avoid the code duplication, something like this:
```diff
index 18dd639d..d7b8427a 100644
--- a/pkg/client/fake/client.go
+++ b/pkg/client/fake/client.go
@@ -399,10 +399,11 @@ func (t versionedTracker) update(gvr schema.GroupVersionResource, obj runtime.Ob
        }

        if t.withStatusSubresource.Has(gvk) {
-               if isStatus { // copy everything but status and metadata.ResourceVersion from original object
-                       if err := copyNonStatusFrom(oldObject, obj); err != nil {
-                               return fmt.Errorf("failed to copy non-status field for object with status subresouce: %w", err)
+               if isStatus { // copy status from object into oldObject, then set object to a copy of oldObject
+                       if err := copyStatusFrom(obj, oldObject); err != nil {
+                               return fmt.Errorf("failed to copy the status for object with status subresource: %w", err)
                        }
+                       obj = oldObject.DeepCopyObject().(client.Object)
                } else { // copy status from original object
                        if err := copyStatusFrom(oldObject, obj); err != nil {
                                return fmt.Errorf("failed to copy the status for object with status subresource: %w", err)
```

(and delete `copyNonStatusFrom`)

Reply:
Yeah, I can do that @alvaroaleman.  I fear we may have this issue with client.Update() using the method to include status with the update there in another issue.  I can fix this and push up the changes and we can discuss on the other issue.

Reply:
This is going to fail lint because the function is not used.  I can add the lint override to not flag this if that is what we want to do.

Reply:
Why not remove the func?

Reply:
I can but didn't want to arbitrarily start deleting code.  I'll delete it.